documentation.md

```diff
@@ -0,0 +1,521 @@
+
+In addition to the pointers, lengths of sub-segments of linked lists are stored to maintain the state of each linked lists, which is not fully defined by the four pointer and metadata RAMs only:
+
+- _rsv_len_: Holds the number of extended cells that have been reserved but have not received any response yet.
+- _rsp_len_: Holds the number of active extended cells.
+
+<figure class="image">
+  <img src="https://i.imgur.com/gj5IPNo.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: Example of rsv_len and rsp_len</figcaption>
+</figure>
+
+Additionally, another length is combinatorically inferred:
+
+- _rsp_len_after_out_: Determines what will be _rsp_len_, considering the release of responses but not the acquisition of new data. This signal is helpful in many regular and corner cases as it helps to manage the latency cycle at the output.
+
+#### Extended cell state
+
+##### Internal counters
+
+For each extended RAM cells, additional memory is dedicated to maintaining the extended RAM cell state:
+
+- _rsv_cnt_: Counts the number of responses that are still awaited in the burst. When reserving an extended cell, this counter is set to the address request burst length. The counter is decremented every time a burst response is transmitted from this extended RAM cell to the requester.
+- _rsp_cnt_: Counts the number of responses that are currently stored in the extended cell. The counter is incremented every time a response is acquired and decremented every time a response is released.
+- _burst_len_: Stores the burst length of the address request corresponding to this extended RAM cell.
+
+The following relationship always holds:
+$$burst\_len \ge rsv\_cnt + rsp\_cnt$$
+
+Additionally, the number of released responses in a burst is given by:
+$$burst\_len - rsv\_cnt - rsp\_cnt$$
+
+And the number of already released responses in a burst is given by:
+$$burst\_len - rsv\_cnt$$
+
+An extended cell is said to be valid (_ram_v_) (in a terminology like a cache line for instance) if _rsv_cnt_ is non-zero or _rsp_cnt_ is non-zero.
+
+##### Elementary cell offset
+
+These three elements maintain full information of the extended cell state and give the offset of an elementary RAM cell inside an extended cell:
+
+- The offset for data output is given by $tail\_burst\_len - tail\_rsv\_cnt - tail\_rsv\_cnt$, which corresponds to the number of burst responses already released. The offset may be taken from the _pre_tail_ instead of the _tail_ in the cases described further above.
+- The offset for data input is given by $rsp\_burst\_len - rsp\_rsv\_cnt$, which corresponds to the number of burst responses already acquired (regardless of whether they have been already released).
+
+#### Linked list detailed operation
+
+A pointer _pA_ is said to be _piggybacked with_ another pointer _pB_ when we impose that _pA_ takes the same value as _pB_. Typically, this happens when _pA_ needs to be updated but has to stay behind or equal to _pB_.
```

Comment:
I think this doesn't fully make sense. Do you want to say "we impose that _pA_ is modified together with _pB_".

Reply:
Yes, but I want to insist on the fact that they may not move. I gave another try, let me know if it's better!